### PR TITLE
tikv/2pc:prewrite primary first when skip_contraint_check is true

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
     This PR `prewrite` primary first when `skip_contraint_check` is true.  If we do not do this, the following case may happened:
1.  In transaction 1, we parallel send `prewrite(a,b,start_ts)` to transaction, here `a` is the primary while `tikv` received the secondary `b`'s request first, it marked with `lock(b,start_ts,primary=a)`.
1.   In transaction 2,  `tikv` find a lock of `b(start_ts)`, it send a `rollback(a,start_ts)` to primary key `a`.
2.   In transaction 2, `tikv` receive a `rollback(a,start_ts)` on the primary, it marked with `write(a,rollback,start_ts,start_ts)`
3.  In transaction 1, `tikv` receive a `prewrite(a,start_ts)` with `skip_contraint_check=true`, it skipped to check the latest commit version, and marked with `lock(a,start_ts),default(a,start_ts)`
4. In transaction 1, `tikv` receive a `commit(a,start_ts, commit_ts)`, since `lock(a,start_ts)` exist, it clear the  `lock(a,start_ts)` and marked with a `write(a,put,start_ts,commit_ts)`

Then there are two commit version with same `start_ts` in the `tikv`, and it caused the disaster.

@zhangjinpeng1987 @disksing @hhkbp2 PTAL

### CODE REVIEW

**Comment by disksing** [comment]
LGTM. btw need to consider how to cover it in test later.

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -323,17 +324,18 @@ func (c *twoPhaseCommitter) keySize(key []byte) int {
 	return len(key)
 }
 
+func (c *twoPhaseCommitter) skipCheckForWrite() bool {
```
Comment (shenli) [inline store/tikv/2pc.go]: Could we add a field in twoPhaseCommitter like SkipConstraintCheck? So we do not need to access GetOption every time.

**Comment by shenli** [comment]
LGTM

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -141,14 +142,17 @@ func newTwoPhaseCommitter(txn *tikvTxn) (*twoPhaseCommitter, error) {
...
-		keys:      keys,
-		mutations: mutations,
-		lockTTL:   txnLockTTL(txn.startTime, size),
-		priority:  getTxnPriority(txn),
+		store:             txn.store,
+		txn:               txn,
+		startTS:           txn.StartTS(),
+		keys:              keys,
+		mutations:         mutations,
+		skipCheckForWrite: skip && ok,
```
Comment (zhangjinpeng1987) [inline store/tikv/2pc.go]: should be ok && skip?

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**